cluster/addons/dns/coredns.yaml.base

```diff
@@ -74,23 +74,26 @@ metadata:
   name: coredns
   namespace: kube-system
   labels:
-    k8s-app: coredns
+    k8s-app: kube-dns
```

Comment:
Why changing label to kube-dns?

Reply:
The dns-autoscaler specifically looks for the label "kube-dns".
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/autoscaling/dns_autoscaling.go#L40

Also, this keeps in sync with the manifest in kubeadm.

Reply:
@rajansandeep But this label might cause confusion when using CoreDNS. How about changing this to `cluster-dns` for both kube-dns and CoreDNS and change other places accordingly. @kubernetes/sig-cluster-lifecycle-pr-reviews WDYT?

Reply:
> How about changing this to cluster-dns for both kube-dns and CoreDNS

Changing label for kube-dns is much more risky because lots of existing clusters have kube-dns pods running with this existing label. Upon master upgrade, kube-dns service update will happen in one shot but kube-dns deployment will be rolling updated, the actual number of kube-dns backends will be drop to 0 at the beginning and will likely cause outage.

Reply:
Agreed. :100: :+1:

Reply:
@rajansandeep If we are setting the labels wrong when deploying coredns in kubeadm, I would much rather that we fix the label there. Would it be possible to change the label going forward and provide backward compatibility for the old kube-dns label within the autoscaler itself?

Reply:
It looks like https://github.com/kubernetes-incubator/cluster-proportional-autoscaler/issues/9 might be relevant in this case.